docs/ldml/tr35-keyboards.md

```diff
@@ -308,6 +308,47 @@ For extensibility, the `<special>` element will be allowed at nearly every level
 
 See [Element special](tr35.md#special) in Part 1.
 
+## Normalization
+
+Normalization will not typically be the responsibility of the keyboard author, rather this will be managed by the implementation.
+The implementation will apply normalization as appropriate when matching transform rules and `<display>` value matching.
+Output from the keyboard, following application of all transform rules, will be normalized to implementation- or application-requested form.
+
+The attribute value `normalization="disabled"` can be used to indicate that no automatic normalization happens.
+Using this setting is not recommended. See [`<settings>`](#element-settings) for details.
+
+Input source files may be in any normalization format, however, authors should be aware of two areas where normalization affects keyboard operation: that of transform matching, and that of output.
+
+### Normalization and Transform Matching
+
+Regardless of the normalization form in the keyboard source file or in the edit buffer context, transform matching will be performed using **NFD**. For example, all of the following transforms will match the input strings `è̠`, whether the input is U+00E8 U+0320, U+0065 U+0320 U+0300, or U+0065 U+0300 U+0320.
+
+```xml
+<transform from="e\u{0300}\u{0320}" /> <!-- Non-NFD but will match -->
+<transform from="e\u{0320}\u{0300}" />
+<transform from="\u{00E8}\u{0320}" /> <!-- è + U+0320 -->
+```
+
+#### Normalization and Markers
+
+A special issue occurs when markers are involved. The markers may be reordered along with characters.
+
+For example, the string `e\u{0300}\u{0320}` will be reordered in NFD to `e\u{0320}\u{0300}`.
+
+If a marker is included, then `e\u{0300}\m{marker}\u{0320}` will be reordered to `e\m{marker}\u{0320}\u{0300}`. The principle is that the marker stays glued to the following character, in this case the U+0320.
+
+TODO
+
+- If the marker is at the end of the string.
```

Comment:
The natural implication would be that...

```suggestion
For a marker immediately before the current location of the text output caret, the marker stays glued to the caret's location until a new character is emitted, at which point the marker attaches to that new character.  It does not attach to any text located after the caret.
```

Of course, the question is if that is reasonable implementation-wise, rather than adding unwanted complications.  I imagine the latter is the case, given that this is a "TODO"?

Reply:
The normalization or transforms aren't dynamic, that is, processing happens after a keystroke.  So yes, the marker would be attached to the end of string *for purposes of normalization*.  When a new character is added to the context, that becomes the new attachment point.